### PR TITLE
feat(engine): pause protocol execution if door is opened

### LINKED ISSUE #9125: feat(engine): pause protocol execution if front door is opened
## Overview

As a JSONv6 protocol operator with my robot's "pause execution if door is open" setting turned on, I'd like my protocol run to automatically pause if my robot's front door is opened.

## Acceptance criteria

If the door pause setting is enable:

- [ ] If the front-door is opened while a protocol run is ongoing, the engine and hardware API are paused
- [ ] If the front-door is opened during protocol setup (i.e. LPC), the engine continues to process commands

## Implementation notes

- The behavior is currently implemented in the `LegacyContextPlugin` for PAPIv2 protocols
    - We can probably move at least aspects of that implementation over to the engine proper
- We can check the command store's `started_at` field to know if the run has been started and if we should pause due to a door open
- Care should be taken to make sure this interacts cleanly with #7923
    - This is also already a concern of `LegacyContextPlugin`

### PR DESCRIPTION
**Pull request by jbleon95**
# Overview
Closes #9125 

Adds pausing protocol execution if a door state hardware notification is detected to protocol engine. Protocol Engine hardware action and handlers renamed to reflect dealing with door events only, along with simplification of the door change action.

# Changelog
- Pauses protocol execution upon `DoorStateNotification`
- Refactor and rename PE HardwareEvent names
- Remove legacy action handling

# Review requests
# Risk assessment
Low, mostly refactoring and renaming with only small and contained functionality added

### CODE REVIEW

File: api/src/opentrons/protocol_engine/execution/door_watcher.py [hunk 1]
```diff
@@ -71,22 +80,26 @@ def _handle_hardware_event(self, event: HardwareEvent) -> None:
...
         """Dispatch an action into self._action_dispatcher.
 
         This must run in the event loop that owns self._action_dispatcher, for safety.
 
-        Defined as an async function so we can use this with
+        Defined as an async function, so we can use this with
         run_coroutine_threadsafe(), which lets us block until
         the dispatch completes.
         """
+        if self._state_store.commands.get_is_running():
```
Comment (mcous) [inline api/src/opentrons/protocol_engine/execution/door_watcher.py]: I think this is missing an additional check for `action.door_state == DoorState.OPEN`

File: api/tests/opentrons/protocol_engine/execution/test_door_watcher.py [hunk 2]
```diff
@@ -60,21 +73,27 @@ async def test_event_forwarding(
...
 
+    decoy.when(state_store.commands.get_is_running()).then_return(True)
+
     subject.start()
 
     captured_handler = cast(HardwareEventHandler, handler_captor.value)
 
     input_event = DoorStateNotification(new_state=DoorState.OPEN)
-    expected_action_to_forward = HardwareEventAction(input_event)
+    expected_action_to_forward = DoorChangeAction(DoorState.OPEN)
```
Comment (mcous) [inline api/tests/opentrons/protocol_engine/execution/test_door_watcher.py:83]: We should add a test that `pause` is _not_ called (`times=0`) if `DoorState.CLOSED`

**Review by mcous — DISMISSED** [review]

File: api/src/opentrons/protocol_engine/execution/door_watcher.py [hunk 3]
```diff
@@ -71,22 +81,29 @@ def _handle_hardware_event(self, event: HardwareEvent) -> None:
...
 
         This must run in the event loop that owns self._action_dispatcher, for safety.
 
-        Defined as an async function so we can use this with
+        Defined as an async function, so we can use this with
         run_coroutine_threadsafe(), which lets us block until
         the dispatch completes.
         """
+        if (
+            self._state_store.commands.get_is_running()
```
Comment (mcous) [inline api/src/opentrons/protocol_engine/execution/door_watcher.py:104]: Crap, my bad, missed one more condition we need: `self._state_store.config.block_on_door_open`

**Review by mcous — COMMENTED** [review]
LGTM, pending robot testing. Will approve once someone can work through these smoke tests

JSONv6, door pause setting off:

- [x] Door open does nothing during LPC
- [x] Door close does nothing during LPC
- [x] Door open does nothing during protocol run
- [x] Door close does nothing during protocol run

JSONv6, door pause setting on:

- [x] Door open does nothing during LPC
- [x] Door close does nothing during LPC
- [x] Door open during protocol run pauses
- [x] {"actionType": "play"} errors while door is still open
- [x] Door close followed by {"actionType": "play"} resumes the run

PAPIv2, door pause setting off:

- [x] Door open does nothing during LPC
- [x] Door close does nothing during LPC
- [x] Door open does nothing during protocol run
- [x] Door close does nothing during protocol run

PAPIv2, door pause setting on:
- [x] Door open does nothing during LPC
- [x] Door close does nothing during LPC
- [x] Door open during protocol run pauses
- [x] {"actionType": "play"} errors while door is still open
- [x] Door close followed by {"actionType": "play"} resumes the run

Edit: All above tested by @SyntaxColoring.

**Review by SyntaxColoring — APPROVED** [review]
I haven't looked at the code, but I've run through @mcous's tests, and this appears to be working as specified.

**Review by mcous — APPROVED** [review]
Approving based on @SyntaxColoring's testing

**Pull request merged**